### ISSUE TITLE
Get rid of `fswatch`

### ISSUE THREAD

**Issue by rodja** [zauberzeug/livesync#1]
Instead of using the bash command `fswatch` to monitor for changes it would be cleaner and more platform-independend to use https://pypi.org/project/watchfiles/ or a similar python lib.

Tasks:

- [x] Replace fswatch with watchfiles
- [x] Use .syncignore for `awatch()`
- [x] specify dependencies "watchfiles" and "pathspec"
- [x] Fix "RuntimeError: Already borrowed" after pressing Ctrl-C

**Comment by falkoschindler** [comment]
@rodja Since you wrote the setup.py I'd like to assign the two new dependencies for the PyPI package to you.

Maybe you also have an idea about the RuntimeError. I'm out of ideas. But at least it's working regardless of the "graceless" shutdown.

**Comment by rodja** [comment]
The RuntimeError "Already borrowed" is strange. Watchfiles has a recent issue which is quite similar: https://github.com/samuelcolvin/watchfiles/issues/177. They filed it to pyo3 where it got closed because it just disappears with a new python version: https://github.com/PyO3/pyo3/issues/2525. But in our case the exception also occurs with python 3.10.5.

The documentation of Watchfiles states in [awatch](https://watchfiles.helpmanual.io/api/watch/#watchfiles.awatch):

> Unlike [watch](https://watchfiles.helpmanual.io/api/watch/#watchfiles.watch) KeyboardInterrupt cannot be suppressed by awatch so they need to be caught where asyncio.run or equivalent is called.

But if we handle the KeyboardInterrupt around `asyncio.run` this internal error is printed:

```
unhandled exception during asyncio.run() shutdown
task: <Task finished name='Task-2' coro=<Folder.watch() done, defined at /Users/rodja/Projects/livesync/livesync/folder.py:58> exception=RuntimeError('Already borrowed')>
Traceback (most recent call last):
  File "/Users/rodja/Projects/livesync/livesync/folder.py", line 59, in watch
    async for changes in watchfiles.awatch(self.local_dir, stop_event=self._stop_watching,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/watchfiles/main.py", line 222, in awatch
    with RustNotify([str(p) for p in paths], debug, force_polling, poll_delay_ms) as watcher:
RuntimeError: Already borrowed
```

Unfortunately the RuntimeError occurs before we reach the KeyboardInterrupt. Else we could call `stop_watching` on each folder.

**Comment by rodja** [comment]
I think we can quite safely drop the `RuntimeException: Already borrowed` for now: https://github.com/zauberzeug/livesync/commit/f93db3f245cc32573869048c80b7329fa057bf08

**Comment by rodja** [comment]
Available in latest [0.2.1](https://github.com/zauberzeug/livesync/releases/tag/0.2.1) release.
